### ISSUE TITLE
Unable to copy-mode -M in top line, scrolling starts instead

### ISSUE THREAD

**Issue by kkovacs** [tmux/tmux#2104]
### Issue description

With just the following `.tmux.conf`:

```
bind -n DoubleClick1Pane copy-mode -M
set -g mouse
```

Double-clicking anywhere in the pane works, except for the top line (1st line of pane). Double-clicking the top line also starts the selection, but **immediately** starts a scrolling back too. (Which is unexpected when one just wanted to enter copy-mode.)

Ps.: Just FYI, what I'm **ultimately** trying to do (and it worked well until about a few months ago) is to emulate how most terminal software works: starting a selection by selecting the current word on double-click:

```
bind -n DoubleClick1Pane select-pane \; copy-mode -M \; send-keys -X select-word
```

### Required information

Please provide the following information:

* tmux version (`tmux -V`).

tmux 3.0a

* Platform (`uname -sp`).

Darwin i386.

iTerm2 terminal

* $TERM inside and outside of tmux (`echo $TERM`).

screen-256color (inside and outside)

* Logs from tmux (`tmux kill-server; tmux -vv new`).

(empty log)

**Comment by nicm** [comment]
Why are you using `-M` if you just want to enter copy mode?

**Comment by kkovacs** [comment]
First of all, thanks for the tip, it does fix the given issue.

But, the `-M` is used because in the `copy-mode -M \; send-keys -X select-word` situation, without the `-M` on the `copy-mode`, when following up the double-click with a mouse drag selection starts from the **center* of the word initially selected by `select-word`, while **with** `-M` it keeps the whole word selected (as it works in most software: terminals, browsers, etc).

Basically the behaviour I'm trying to do is:

1. mouse double-click on a word selects the whole word
2. continue dragging with the mouse, either backwards or forwards, continues the selection.

It would be great to get some guidance how to achieve this. :)

Keep up the great work!

**Comment by nicm** [comment]
I don't see this, with and without -M work the same for me except that -M sets the dragging flag which means the top line will scroll. Do you see the same if you build tmux from master?

**Comment by nicm** [comment]
tmux is slightly different from xterm here in fact. In tmux, double clicking to select a word and then later dragging continues the selection as a word selection (that is, the word selection is sticky). In xterm, double clicking to select a word and then dragging creates an entirely new selection from the point of the new selection - a word selection is only done if you do not release the mouse button on the second click and instead drag with it.

Anyway, you will need master or 3.1-rc for word selections.

**Comment by kkovacs** [comment]
Thank you, having this feature will be great, I was not aware that 3.1 brings changes in this, thanks for working on it! 

I'll try to build from `master` now!

**Comment by tigrabpbar** [comment]
I can confirm that on tag `3.1` word selection works perfectly. Word selection being sticky is brilliant! \o/

One minor related issue I notice though: when I start with `word-select` as above, then using the key `o` to switch between the two ends of the selection, then often the start of the selection gets "reset" to the starting position, even if it was moved already.

To reproduce step by step:

1. `:bind -n DoubleClick1Pane select-pane \; copy-mode \; send-keys -X select-word`
2. Double-click a word
3. Do some mouse drag to expand selection
4. Use `o` to jump to the **start** of the selection
5. Move the start of the selection a bit
6. Use 'o' again to jump to the **end** of the selection
7. Bug: start of the selection resets back to where it was between 4 and 5 (not stay as it was between 5 and 6).

Thanks for all the work you do on `tmux`!
